crates/bevy_render/src/render_resource/pipeline.rs

```diff
@@ -128,6 +128,34 @@ pub struct VertexBufferLayout {
     pub attributes: Vec<VertexAttribute>,
 }
 
+impl VertexBufferLayout {
+    /// Creates a new densely packed [`VertexBufferLayout`] from an iterator of vertex formats.
+    /// Iteration order determines the `shader_location` and `offset` of the VertexAttributes.
+    /// The first iterated item will have a `shader_location` and `offset` of zero.
+    /// The `array_stride` is the sum of the size of the iterated VertexFormats (in bytes).
+    pub fn from_vertex_formats<T: IntoIterator<Item = VertexFormat>>(
+        step_mode: VertexStepMode,
+        vertex_formats: T,
+    ) -> Self {
+        let mut array_stride = 0;
+        let mut attributes = Vec::new();
+        for (shader_location, format) in vertex_formats.into_iter().enumerate() {
+            attributes.push(VertexAttribute {
+                format,
+                offset: array_stride,
```

Comment:
I had to read this and then backtrack to understand why `array_stride` was used for the offset. I feel like I'd prefer to call the variable `offset`and then assign it as the `array_stride` value just below in the `VertexBufferLayout`.

Reply:
Good call